django_extensions/management/commands/drop_test_database.py

```diff
@@ -82,26 +83,61 @@ def handle(self, *args, **options):
         verbosity = options["verbosity"]
         if options['interactive']:
             confirm = input("""
-You have requested to drop the test database.
+You have requested to drop all test databases.
 This will IRREVERSIBLY DESTROY
-ALL data in the database "%s".
+ALL data in the database "{db_name}"
+and all cloned test databases generated via
+the "--parallel" flag (these are sequentially
+named "{db_name}_1", "{db_name}_2", etc.).
 Are you sure you want to do this?
 
-Type 'yes' to continue, or 'no' to cancel: """ % (database_name,))
+Type 'yes' to continue, or 'no' to cancel: """.format(db_name=database_name))
         else:
             confirm = 'yes'
 
         if confirm != 'yes':
             print("Reset cancelled.")
             return
 
+        def get_database_names(formatter):
+            """
+            Return a generator of all possible test database names.
+            e.g., 'test_foo', 'test_foo_1', test_foo_2', etc.
+
+            formatter: func returning a clone db name given the primary db name
+            and the clone's number, e.g., 'test_foo_1' for mysql/postgres, and
+            'test_foo_1..sqlite3' for sqlite (re: double dots, see comments).
+            """
+            yield database_name
+            yield from (formatter(database_name, n) for n in count(1))
+
         if engine in SQLITE_ENGINES:
+            # By default all sqlite test databases are created in memory.
+            # There will only be database files to delete if the developer has
+            # specified a test database name, which forces files to be written
+            # to disk.
+
+            logging.info("Unlinking %s databases" % engine)
+
+            def format_filename(name, number):
+                filename, ext = os.path.splitext(name)
+                # Since splitext() includes the dot in 'ext', the inclusion of
+                # the dot in the format string below is incorrect and creates a
+                # double dot. Django makes this mistake, so it must be
+                # replicated here. If fixed in Django, this code should be
+                # updated accordingly.
+                # Reference: https://code.djangoproject.com/ticket/32582
+                return '{}_{}.{}'.format(filename, number, ext)
+
             try:
-                logging.info("Unlinking %s database" % engine)
-                if os.path.isfile(database_name):
-                    os.unlink(database_name)
+                for db_name in get_database_names(format_filename):
+                    if not os.path.isfile(db_name):
+                        break
```

Comment:
Would `break` or `continue` be more appropriate here.... ?

(eg: say I deleted `foo_bar_test_2.db` myself but not `foo_bar_test_3.db`...)

Reply:
The issue is that a developer can supply any value to `--parallel`, and Django will create that many test databases up to a maximum of the number of TestCase classes that exist. I could create 1000 databases with `manage.py --parallel 1000 --keepdb`. The `drop_test_database` code has no knowledge of the value last used with `--parallel`, so some other strategy must exist to halt the loop that looks for test databases to delete. 

The most reasonable strategy to my mind is checking sequential existence, since the only guarantee we can rely on from Django's process is that the test database names are sequentially numbered. That does leave the caveat you point out that a developer could manually delete some test database in the middle and then run `drop_test_database`, but to me it seems that would be the developer's responsibility to clean up, since it's an odd thing to do.

Alternatively, an arbitrarily high upper limit on the possible number could be hard-coded into the loop, and the loop could always run that number of times, checking all database names. This approach to me is inferior, since it relies on guessing some value assumed to be higher than any developer would use, and is slower since it would need to run that number of database queries (or file existence checks in the case of sqlite) every time the command is run. And, despite the downsides, it still leaves its own caveat since some developer could always create a larger number of test database than the maximum we set on this loop and those won't get deleted.

If you do want the second strategy, let me know what value cap you want to use, and I'll implement that. Or if you have an idea for a different strategy to solve this problem, let me know. But my current recommendation would be what's implemented, and perhaps if you prefer, I could add something to the docs warning that the command may not remove everything if a developer manually deletes any test databases first.